src/util/apparmor.c

```diff
@@ -93,3 +105,453 @@ int bus_apparmor_dbus_supported(bool *supportedp) {
         *supportedp = supported;
         return 0;
 }
+
+static bool is_apparmor_enabled(void) {
+        int ret;
+        bool enabled;
+
+        ret = bus_apparmor_dbus_supported(&enabled);
+        if (ret || !enabled)
+                return false;
+
+        ret = bus_apparmor_is_enabled(&enabled);
+        if (ret || !enabled)
+                return false;
```

Comment:
You are calling this on every message transaction. I don't think we can do that, that is quite slow. I think it is better to just check `registry->bustype` and consider it disabled if it is NULL.